lib/Catalyst/DispatchType/Chained.pm

```diff
@@ -378,7 +378,10 @@ sub register {
 
     $action->attributes->{PathPart} = [ $encoded_part ];
 
-    unshift(@{ $children->{$encoded_part} ||= [] }, $action);
+    # "Reversed" sort: best actions should be at the end
+    $children->{$encoded_part} = [
```

Comment:
As "semi-update" maybe is reasonable to add method `add_part_action` with default "old" realization:
```
unshift(@{ $children->{$encoded_part} ||= [] }, $action);
```